### PR TITLE
Fix concurrency issues occuring when onboarding multiple devices in parallel

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: authorization/client/userDevices.go [hunk 1]
```diff
@@ -125,6 +125,50 @@ func (d *UserDevicesManager) GetUserDevices(ctx context.Context, userID string)
...
+		return false
+	}
+	for _, id := range userDevices[userID] {
+		if id == deviceID {
+			return true
+		}
+	}
+	return false
+}
+
```
Comment (jkralik) [inline authorization/client/userDevices.go]: remove me

File: coap-gateway/service/server.go [hunk 2]
```diff
@@ -328,7 +328,9 @@ func (server *Server) ServiceRequestContext(userID string) (context.Context, con
 	if err != nil {
 		return nil, nil, err
 	}
-	ctx, cancel := context.WithTimeout(kitNetGrpc.CtxWithUserID(kitNetGrpc.CtxWithToken(server.ctx, serviceToken.AccessToken), userID), server.RequestTimeout)
+	ctx := kitNetGrpc.CtxWithUserID(kitNetGrpc.CtxWithToken(server.ctx, serviceToken.AccessToken), userID)
+	cancel := func() {}
+	//ctx, cancel := context.WithTimeout(kitNetGrpc.CtxWithUserID(kitNetGrpc.CtxWithToken(server.ctx, serviceToken.AccessToken), userID), server.RequestTimeout)
```
Comment (jkralik) [inline coap-gateway/service/server.go]: remove me

File: authorization/client/userDevices.go [hunk 3]
```diff
@@ -125,6 +125,28 @@ func (d *UserDevicesManager) GetUserDevices(ctx context.Context, userID string)
 	return devs, nil
 }
 
+// UpdateUserDevices updates and returns devices which belows to user.
```
Comment (ondrejtomcik) [inline authorization/client/userDevices.go]: typo - belows => belongs

File: bundle/run.sh [hunk 4]
```diff
@@ -34,6 +34,7 @@ export LISTEN_FILE_CERT_KEY_NAME="$GRPC_INTERNAL_CERT_KEY_NAME"
 
 export MONGODB_URL="mongodb://localhost:$MONGO_PORT"
 export MONGODB_URI="mongodb://localhost:$MONGO_PORT"
+export MONGODB_MAX_POOL_SIZE="$RESOURCE_AGGREGATE_MONGO_MAX_PARALLEL_QUERIES"
```
Comment (ondrejtomcik) [inline bundle/run.sh:37]: Please document this env variable in bundle documentation with explanation.

**Review by ondrejtomcik — APPROVED** [review]

**Pull request merged**